content/blog/customer-Q-and-A.md

```diff
@@ -0,0 +1,758 @@
+
+**A:**
+The kn-workflow gen-manifest (the CLI to generate the manifests) expects only one sw.yaml or sw.json file.
+If there is more than a single file, the CLI will fail with:
+
+```bash
+❌ ERROR: generating manifests: ❌ ERROR: multiple SonataFlow definition files found
+```
+
+The reason for that is that the main workflow is translated into a SonataFlow CR that represents a single workflow.
+
+If there is a need to add additional subflows, they can be placed in their own directory and referred by the kn-workflow gen-manifest CLI using the --subflows-dir flag
+
+In that case, subflows will have to be placed under
+src/main/resources/subflows
+
+</details>
+
+<details>
+<summary><strong>Q: How can I lifecycle workflow / templates if there is a new version available?</strong></summary>
+
+**A:** It really depends on how the workflow is designed and implemented.
+A few options:
+
+- declarative configuration which is synced via GitOps might be best if PR are issued to projects, using the template. The responsible developer team. or app owners should be responsible to implement the change and test it. Tool like renovatebot or dependabot might help here.
+- update in workflows which are implemented in an imperative approach might need "update workflows" who help to bring out the changes from one version to the next one. Example could be database changes / patches to new api versions, etc.
+
+In any case, with RHDH and Orchestrator you should be equipped to analyse which app is installed where in which version and then be able to plan next steps.
+
+</details>
+
+<details>
+<summary><strong>Q: Can we use the Quarkus extension for building workflow images or does the build.sh script more stuff then just building the image in the standard way?</strong></summary>
+
+**A:** The build.sh script isn't a must. It offers an opinionated method to build workflows with specific Quarkus extensions that are recommended for production and enables the persistence.
+
+The core part of building workflow is using the openshift-serverless-logic builder image as referenced from https://github.com/rhdhorchestrator/orchestrator-demo/blob/main/docker/osl.Dockerfile#L4
+
+To build the workflow image it is enough to run (with all or part of the extensions, depends on your needs):
+
+```bash
+podman build \
+ -f ../../orchestrator-demo/docker/osl.Dockerfile \
+ --tag $TARGET_IMAGE \
+  --platform linux/amd64 \
+  --ulimit nofile=4096:4096 \
+  --build-arg QUARKUS_EXTENSIONS="\
+    org.kie:kie-addons-quarkus-persistence-jdbc:9.102.0.redhat-00005,\
+    io.quarkus:quarkus-jdbc-postgresql:3.8.6.redhat-00004,\
+    io.quarkus:quarkus-agroal:3.8.6.redhat-00004, \
+    io.quarkus:quarkus-smallrye-reactive-messaging-kafka" \
+  --build-arg MAVEN_ARGS_APPEND="\
+    -DmaxYamlCodePoints=35000000 \
+    -Dkogito.persistence.type=jdbc \
+    -Dquarkus.datasource.db-kind=postgresql \
+    -Dkogito.persistence.proto.marshaller=false" \
+    .
+```
```

Comment:
not a big fan of having this, @masayag WDYT? I would remove it as the parameters may change over time